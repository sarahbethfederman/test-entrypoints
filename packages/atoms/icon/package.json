{
  "name": "@auscred/icon",
  "version": "0.0.1",
  "source": "src/Icon.tsx",
  "main": "dist/Icon.js",
  "types": "dist/Icon.d.ts",
  "directories": {
    "lib": "lib"
  },
  "keywords": [],
  "author": "Lendi Devs",
  "license": "ISC",
  "devDependencies": {
    "jest": "^23.1.0",
    "@lui/story-utils": "^1.0.0",
    "@storybook/addon-knobs": "^3.4.8",
    "@types/storybook__addon-knobs": "^3.3.1"
  },
  "scripts": {
    "build": "node ./src/build.js && tsc -p ./tsconfig.json"
  },
  "dependencies": {
    "svg-sprite": "^1.4.0"
  }
}
