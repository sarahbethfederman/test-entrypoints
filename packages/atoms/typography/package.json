{
  "name": "@auscred/typography",
  "version": "0.0.1",
  "main": "dist/Typography.js",
  "types": "dist/Typography.d.ts",
  "files": [
    "dist"
  ],
  "keywords": [],
  "author": "Lendi Devs",
  "license": "ISC",
  "peerDependencies": {
    "@types/react": "^16.4.0",
    "react": "^16.4.0",
    "styled-components": "^3.3.3"
  },
  "scripts": {
    "build": "tsc --pretty"
  }
}
