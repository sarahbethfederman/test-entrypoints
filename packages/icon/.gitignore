src/icons-compiled/*
src/index.ts