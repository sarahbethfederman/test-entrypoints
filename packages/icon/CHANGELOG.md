## [7.1.0] - 2019-05-02 - [HUB-126] (https://creditandfinance.atlassian.net/browse/HUB-126)
### Added
- Added the following new icons: file-upload & FileDownload as an alias to GetApp.

## [7.0.1] - 2019-05-02 - [HUB-132] (https://creditandfinance.atlassian.net/browse/HUB-132)
### Changed
- Fix MS's Edge Icon rendering issue

## [7.0.0] - 2019-04-30 - [HUB-54] (https://creditandfinance.atlassian.net/browse/HUB-54)
### Deleted
- Remove the CJS bundles.

## [6.0.0] - 2019-04-16 - [HUB-49](https://creditandfinance.atlassian.net/browse/HUB-49)
### Changed
- Remove Left and Right Icons.

## [5.3.1] - 2019-04-08 - [HUB-71](https://creditandfinance.atlassian.net/browse/HUB-71)
### Changed
- Added styling to the icon container to align text and icon baseline when using inline icons.

## [5.3.0] - 2019-04-01 - [HUB-53](https://creditandfinance.atlassian.net/browse/HUB-53)
### Added
- Added the following new icons: attach-file, assignment-returned, autorenew, assessment, save, credit-card, assignment-return, people, more-horiz, done-all, assignment-turned-in, error2, search, date-range, assignment, open-with, dashboard, timeline, gavel, trending-flat, report-problem, trending-down, delete, fullscreen, delete-outline, remove-circle, sentiment-satisfied-alt, launch, edit, add-alert, trending-up, help, history, schedule, room, notification-important, get-app, share, highlight-off, more-vert, favorite-border, hourglass-empty, info2, monetization-on, thumb-up, help-outline, error-outline, add, check2, warning, build, computer, data-usage, alarm, description, stay-current-portrait, power-settings-new, account-balance

## [5.2.0] - 2019-03-15 - [BRAVO-576](https://creditandfinance.atlassian.net/browse/BRAVO-576)
### Added
- Added un-circled warn, info and success icons `info-two success-two warn-two`

## [5.1.0] - 2019-03-06 - [BRAVO-572](https://creditandfinance.atlassian.net/browse/BRAVO-572)
### Added
- added Maintenance icon and PageNotFound icon

## [5.0.1] - 2019-03-05 - [BRAVO-387](https://creditandfinance.atlassian.net/browse/BRAVO-387)
### Added
- added Domain logo

5.0.0 - Breaking change version update
4.4.0 - Bump version for left and right icons
