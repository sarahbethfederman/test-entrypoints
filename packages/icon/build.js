const SVGSpriter = require('svg-sprite');
const fs = require('fs');
const path = require('path');
const mkdirp = require('mkdirp');

const directory = `${__dirname}/src/icons`;

const spritify = new SVGSpriter({
  dest: 'dist',
  mode: {
    symbol: true,
    inline: true,
  },
});

fs.readdir(directory, (err, files) => {
  if (err) {
    console.error(err);
    return;
  }

  files.forEach((file) => {
    if (file.endsWith('.svg')) {
      spritify.add(
        path.resolve(`${directory}/${file}`),
        file,
        fs.readFileSync(`${directory}/${file}`, { encoding: 'utf-8' })
      );
    }
  });

  spritify.compile(function(error, result, data) {
    for (var mode in result) {
      for (var resource in result[mode]) {
        fs.writeFileSync(`../../website/static/sprite.symbol.svg`, result[mode][resource].contents);
      }
    }
  });
});
