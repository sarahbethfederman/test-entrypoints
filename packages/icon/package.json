{
  "name": "@lendi-ui/icon",
  "version": "9.3.0",
  "main": "dist/cjs/index.js",
  "source": "src/index.tsx",
  "types": "dist/types/index.d.ts",
  "files": [
    "dist"
  ],
  "keywords": [],
  "author": "Lendi Devs",
  "license": "ISC",
  "peerDependencies": {
    "@lendi-ui/theme": "^8.0.0",
    "@types/react": "^16.8",
    "@types/styled-components": "^4.1.18",
    "react": "^16.8",
    "styled-components": "^4"
  },
  "devDependencies": {
    "@svgr/core": "^2.2.0",
    "@lendi-ui/typography": "^4.0.0",
    "@lendi-ui/spacing": "^6.0.0"
  },
  "dependencies": {
    "@lendi-ui/color": "^5.0.0"
  },
  "scripts": {
    "build:svg": "node ./build/index.js",
    "build:cjs": "tsc --project tsconfig.cjs.json",
    "build:types": "tsc --project tsconfig.types.json",
    "build": "yarn run build:svg && yarn run build:types && yarn run build:cjs"
  }
}
