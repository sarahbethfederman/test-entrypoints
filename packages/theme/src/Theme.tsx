import * as React from 'react';
import styled, { ThemeProvider, injectGlobal } from 'styled-components';
import { theme } from './theme/lendi';
import { ThemeMap } from './types';

interface ThemeProps {
  kind?: ThemeMap;
  children?: React.ReactNode;
}

// Any cross browser resets should occur here
// tslint:disable-next-line no-unused-expression
injectGlobal`
  html, body {
    margin: 0;
    padding: 0;
  }

  :root {
    --lendi-ui-size: 14px;
  }

  @media (min-width: 36.0625em) {
    :root {
      --lendi-ui-size: 16px;
    }
  }
`;

const Root = styled.div`
  color: ${theme.colors.shade[700]};
`;

export function Theme(props: ThemeProps) {
  const { children } = props;
  const kind = props.kind || theme;

  return (
    <ThemeProvider theme={kind}>
      <Root>{children}</Root>
    </ThemeProvider>
  );
}
