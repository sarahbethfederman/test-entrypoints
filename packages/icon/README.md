# @lendi-ui/icon

## Adding a new icon

Add the SVG into src/icons. Be thoughtful with naming, as this will determine the name of the component. For example, the file `computer-bank` will create a SVG component called `computerBank` (note that the kebab-case filename will become a camelCase component).

To add a new alias, navigate to the ./build/alias.json file and add the key as the original file name (in pascal case) and give it an array with all the alias' to add.

When this package is built, an equivalent react component will be created in src/icons-compiled.

## Installation

```
yarn add @lendi-ui/icon
```

## Usage

```jsx
import { Lock } from '@lendi-ui/icon';

<Lock color="secondary.500" />;
```

## Fill colours not working?

In this case, you will need to add a `fillPath` class to the element in the SVG that needs to be recoloured, this can be determined by inspecting the svg in the browser and changing the fill colour until you find the correct element.

## Properties

```
<PropTable>
  <PropTable.Entry required name="color" type={`string`} description="The Icon color"/>
</PropTable>
```
