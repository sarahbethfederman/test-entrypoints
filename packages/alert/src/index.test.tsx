import * as React from 'react';
import { mount, ReactWrapper } from 'enzyme';
import Theme from '@lendi-ui/theme';
import Alert, { AlertProps } from './index';
import { Wrapper, IconWrapper, HeaderWrapper, HeadingWrapper, ContentWrapper } from './index.style';
import { Info, ErrorOutline, CheckCircleOutline } from '@lendi-ui/icon';
import { Link } from '@lendi-ui/typography';

let element: ReactWrapper<AlertProps>, variant;

function render(props) {
  element = mount(
    <Theme>
      <Alert variant="error" {...props}>
        Error description
      </Alert>
    </Theme>
  );
}

function renderWithComponent(props) {
  element = mount(
    <Theme>
      <Alert variant="error" {...props}>
        Sorry, you may be blocked from logging in. Please <Link href="#">contact us</Link>.
      </Alert>
    </Theme>
  );
}

describe('Alert', () => {
  it('should mount Alert component without heading', () => {
    render({});
    expect(element.find(Alert)).toHaveLength(1);
    expect(element.find(Wrapper)).toHaveLength(1);
    expect(element.find(IconWrapper)).toHaveLength(1);
    expect(element.find(ErrorOutline)).toHaveLength(1);
    expect(element.find(ErrorOutline).props().color).toEqual('error.500');
    expect(element.find(ContentWrapper)).toHaveLength(1);
    expect(element.find(Alert)).toMatchSnapshot();
  });

  it('should mount Alert component with heading', () => {
    const heading = 'Alert label';
    render({ heading });
    expect(element.find(Alert)).toHaveLength(1);
    expect(element.find(Wrapper)).toHaveLength(1);
    expect(element.find(HeaderWrapper)).toHaveLength(1);
    expect(element.find(HeadingWrapper)).toHaveLength(1);
    expect(element.find(IconWrapper)).toHaveLength(1);
    expect(element.find(ErrorOutline)).toHaveLength(1);
    expect(element.find(ErrorOutline).props().color).toEqual('error.500');
    expect(element.find(ContentWrapper)).toHaveLength(1);
    expect(element.find(Alert)).toMatchSnapshot();
  });

  it("it should mount Alert component with variant is 'error'", () => {
    variant = 'error';
    render({ variant });
    expect(element.find(ErrorOutline)).toHaveLength(1);
    expect(element.find(ErrorOutline).props().color).toEqual('error.500');
  });

  it("it should mount Alert component with variant is 'info'", () => {
    variant = 'info';
    render({ variant });
    expect(element.find(Info)).toHaveLength(1);
    expect(element.find(Info).props().color).toEqual('info.500');
  });

  it("it should mount Alert component with variant is 'success'", () => {
    variant = 'success';
    render({ variant });
    expect(element.find(CheckCircleOutline)).toHaveLength(1);
    expect(element.find(CheckCircleOutline).props().color).toEqual('success.500');
  });

  it("it should mount Alert component with variant is 'warn'", () => {
    variant = 'warn';
    render({ variant });
    expect(element.find(ErrorOutline)).toHaveLength(1);
    expect(element.find(ErrorOutline).props().color).toEqual('warn.500');
  });

  describe('test native props and Standard HTML Attributes', () => {
    it('should mount with Aria attributes', () => {
      const ARIA_LABEL = 'testLabel';
      const ARIA_DESCRIBE_BY = 'info';
      render({
        'aria-label': ARIA_LABEL,
        'aria-describedby': ARIA_DESCRIBE_BY,
      });
      const attributes = element.find(Alert).props();
      expect(attributes['aria-label']).toBe(ARIA_LABEL);
      expect(attributes['aria-describedby']).toBe(ARIA_DESCRIBE_BY);
    });
    it('should mount with native props like id, title', () => {
      const TEXT_ID = 'testId';
      const TEXT_TITLE = 'testTitle';
      render({
        id: TEXT_ID,
        title: TEXT_TITLE,
      });
      const attributes = element.find(Alert).props();
      expect(attributes.id).toBe(TEXT_ID);
      expect(attributes.title).toBe(TEXT_TITLE);
    });
  });
});

describe('Alert with rendering Link in children', () => {
  beforeEach(() => renderWithComponent({}));

  it('should render Link component', () => {
    expect(element.find(Link)).toHaveLength(1);
  });
});
