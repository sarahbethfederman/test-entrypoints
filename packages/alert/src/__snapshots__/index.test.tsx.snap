// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert it should mount Alert component with heading 1`] = `
.c7 {
  margin-top: 0;
  margin-bottom: 0;
  font-size: calc(0.75 * var(--lendi-ui-size));
  line-height: calc(16 / 12);
  font-family: 'Open Sans',sans-serif;
}

.c5 {
  margin-top: 0;
  margin-bottom: 0;
  font-size: calc(0.875 * var(--lendi-ui-size));
  line-height: calc(20 / 14);
  font-family: 'Open Sans',sans-serif;
}

.c0 {
  box-sizing: border-box;
  line-height: 1.333;
  white-space: normal;
  border-spacing: 2px;
  text-align: left;
  word-spacing: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  border-radius: 6px;
  padding: var(--lendi-ui-size);
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border: 1px solid #ff1c4a;
  color: #ff1c4a;
  background-color: #fff3f5;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  line-height: 0;
}

.c4 {
  margin-left: calc(0.75 * var(--lendi-ui-size));
}

.c6 {
  margin-top: calc(0.25 * var(--lendi-ui-size));
}

.c3 {
  width: 1em;
  height: 1em;
  vertical-align: middle;
}

.c3 > svg {
  fill: #ff1c4a;
  width: inherit;
  height: inherit;
}

<Alert
  heading="Alert label"
  variant="error"
>
  <styled.div
    variant="error"
    withHeading={true}
  >
    <div
      className="c0"
    >
      <styled.div>
        <div
          className="c1"
        >
          <styled.span>
            <span
              className="c2"
            >
              <Warn
                color="error.500"
              >
                <styled.span
                  color="error.500"
                >
                  <span
                    className="c3"
                    color="error.500"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      viewBox="0 0 24 24"
                      width="1em"
                    >
                      <path
                        d="M11 15h2v2h-2v-2zm0-8h2v6h-2V7zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                      />
                    </svg>
                  </span>
                </styled.span>
              </Warn>
            </span>
          </styled.span>
          <Styled(styled.p)
            size="sm"
          >
            <styled.p
              className="c4"
              size="sm"
            >
              <p
                className="c4 c5"
                size="sm"
              >
                <strong>
                  Alert label
                </strong>
              </p>
            </styled.p>
          </Styled(styled.p)>
        </div>
      </styled.div>
      <Styled(styled.p)
        size="xs"
        withHeading={true}
      >
        <styled.p
          className="c6"
          size="xs"
          withHeading={true}
        >
          <p
            className="c6 c7"
            size="xs"
          >
            Error description
          </p>
        </styled.p>
      </Styled(styled.p)>
    </div>
  </styled.div>
</Alert>
`;

exports[`Alert it should mount Alert component without heading 1`] = `
.c4 {
  margin-top: 0;
  margin-bottom: 0;
  font-size: calc(0.75 * var(--lendi-ui-size));
  line-height: calc(16 / 12);
  font-family: 'Open Sans',sans-serif;
}

.c0 {
  box-sizing: border-box;
  line-height: 1.333;
  white-space: normal;
  border-spacing: 2px;
  text-align: left;
  word-spacing: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  border-radius: 6px;
  padding: var(--lendi-ui-size);
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  border: 1px solid #ff1c4a;
  color: #ff1c4a;
  background-color: #fff3f5;
}

.c1 {
  line-height: 0;
}

.c3 {
  margin-left: calc(0.75 * var(--lendi-ui-size));
}

.c2 {
  width: 1em;
  height: 1em;
  vertical-align: middle;
}

.c2 > svg {
  fill: #ff1c4a;
  width: inherit;
  height: inherit;
}

<Alert
  variant="error"
>
  <styled.div
    variant="error"
    withHeading={false}
  >
    <div
      className="c0"
    >
      <styled.span>
        <span
          className="c1"
        >
          <Warn
            color="error.500"
          >
            <styled.span
              color="error.500"
            >
              <span
                className="c2"
                color="error.500"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  viewBox="0 0 24 24"
                  width="1em"
                >
                  <path
                    d="M11 15h2v2h-2v-2zm0-8h2v6h-2V7zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
              </span>
            </styled.span>
          </Warn>
        </span>
      </styled.span>
      <Styled(styled.p)
        size="xs"
        withHeading={false}
      >
        <styled.p
          className="c3"
          size="xs"
          withHeading={false}
        >
          <p
            className="c3 c4"
            size="xs"
          >
            Error description
          </p>
        </styled.p>
      </Styled(styled.p)>
    </div>
  </styled.div>
</Alert>
`;
